src/amms.py

```diff
@@ -7,297 +7,609 @@
+        client = Client(AUTHENTICATED_RPC_URL)
+
+        for ticker, pool_info in self.TICKERS.items():
+            token_x_acc = client.get_account_info_json_parsed(
+                pool_info["token_x_account"]
+            )
+            token_y_acc = client.get_account_info_json_parsed(
+                pool_info["token_y_account"]
+            )
+
+            if not token_x_acc.value:
+                LOG.warning(f"No value found for X of {self.DEX_NAME}:{ticker}")
+                continue
+
+            if not token_y_acc.value:
+                LOG.warning(f"No value found for Y of {self.DEX_NAME}:{ticker}")
+                continue
+
+            token_x_data = token_x_acc.value.data
+            token_y_data = token_y_acc.value.data
+
+            if not isinstance(token_x_data, ParsedAccount):
+                LOG.warning(f"Unable to parse X of {self.DEX_NAME}:{ticker}")
+                continue
+
+            if not isinstance(token_y_data, ParsedAccount):
+                LOG.warning(f"Unable to parse Y of {self.DEX_NAME}:{ticker}")
+                continue
+
+            token_x_parsed: dict = token_x_data.parsed
+            token_y_parsed: dict = token_y_data.parsed
+
+            token_x_amt = token_x_parsed["info"]["tokenAmount"]["amount"]
+            token_y_amt = token_y_parsed["info"]["tokenAmount"]["amount"]
+
+            token_x_decimals = token_x_parsed["info"]["tokenAmount"]["decimals"]
+            token_y_decimals = token_y_parsed["info"]["tokenAmount"]["decimals"]
+
+            self.pools.append(
+                (
+                    ticker,
+                    {
+                        "token_x_amount": token_x_amt,
+                        "token_y_amount": token_y_amt,
+                        "token_x_decimals": token_x_decimals,
+                        "token_y_decimals": token_y_decimals,
+                    },
+                )
+            )
+
+    def store_pool(self, pool: tuple[str, dict[str, Any]]):
+
+        ticker, pool_info = pool
+
+        with get_db_session() as session:
+            entry = AmmLiquidity(
+                timestamp=self.timestamp,
+                dex=self.DEX_NAME,
+                pair=ticker,
+                market_address="",
+                token_x=pool_info["token_x_amount"],
+                token_y=pool_info["token_y_amount"],
+                token_x_decimals=pool_info["token_x_decimals"],
+                token_y_decimals=pool_info["token_y_decimals"],
+                additional_info="{}",
+            )
+
+            # Add to session
+            session.add(entry)
+
+            #  Commit
+            session.commit()
+
+
+class FluxBeam(Amm):
+    DEX_NAME = "FLUXBEAM"
+
+    async def get_pools(self):  # pylint: disable=W0236
+        LOG.info("Fetching FLUXBEAM pools.")
+        client = AsyncClient(AUTHENTICATED_RPC_URL)
+
+        with open("src/protocols/idls/fluxbeam_pools.json", "rb") as f:
```

Comment:
Where did you get the IDL from? :pray:

Reply:
Yes, pls :pray: 

.. I'd actually prefer to have the list of tokens defined dynamically (only collateralizable or borrowable tokens, maybe filtered by TVL and/or volume). But we can have it fixed/hardcoded for now.